bundles/org.openhab.binding.hdpowerview/src/test/java/org/openhab/binding/hdpowerview/HDPowerViewJUnitTests.java

```diff
@@ -300,93 +352,245 @@ public void shadeResponseIsParsedCorrectly() throws JsonParseException {
+
+        // ==== OK !! secondary at middle, primary below ====
+        shade.setPosition(caps, SECONDARY_ZERO_IS_OPEN, 0).setPosition(caps, PRIMARY_ZERO_IS_CLOSED, 100);
+        shade.setPosition(caps, SECONDARY_ZERO_IS_OPEN, 50).setPosition(caps, PRIMARY_ZERO_IS_CLOSED, 75);
+        assertEquals(new PercentType(50), shade.getState(caps, SECONDARY_ZERO_IS_OPEN));
+        assertEquals(new PercentType(75), shade.getState(caps, PRIMARY_ZERO_IS_CLOSED));
+
+        // ==== IMPOSSIBLE !! primary at middle, secondary below => test the constraining code ====
+        shade.setPosition(caps, SECONDARY_ZERO_IS_OPEN, 0).setPosition(caps, PRIMARY_ZERO_IS_CLOSED, 100);
+        shade.setPosition(caps, PRIMARY_ZERO_IS_CLOSED, 60).setPosition(caps, SECONDARY_ZERO_IS_OPEN, 75);
+        assertEquals(new PercentType(60), shade.getState(caps, PRIMARY_ZERO_IS_CLOSED));
+        assertEquals(new PercentType(60), shade.getState(caps, SECONDARY_ZERO_IS_OPEN));
+
+        // ==== OK !! primary at middle, secondary above ====
+        shade.setPosition(caps, SECONDARY_ZERO_IS_OPEN, 0).setPosition(caps, PRIMARY_ZERO_IS_CLOSED, 100);
+        shade.setPosition(caps, PRIMARY_ZERO_IS_CLOSED, 60).setPosition(caps, SECONDARY_ZERO_IS_OPEN, 25);
+        assertEquals(new PercentType(60), shade.getState(caps, PRIMARY_ZERO_IS_CLOSED));
+        assertEquals(new PercentType(25), shade.getState(caps, SECONDARY_ZERO_IS_OPEN));
+    }
+
+    /**
+     * Print out the positions JSON produced, and the respective output State values, for all possible combinations of
+     * shade position kinds (coordinate systems), shade capabilities, and for a variety of possible position values.
+     *
+     * Note: the code is commented out in order to prevent printing to System.out when the tests are being run in the
+     * automated build system.
+     */
+    @Test
+    public void printOutPositionsJson() {
+        /*
+         * final Gson gson = new Gson();
+         * int[] positions = { 0, 50, 100 };
+         * CoordinateSystem[] coordinates = { SECONDARY_ZERO_IS_OPEN, VANE_COORDS, PRIMARY_ZERO_IS_CLOSED };
+         * for (Capabilities capabilities : ShadeCapabilitiesDatabase.CAPABILITIES_DATABASE) {
+         * ShadePosition shade = new ShadePosition();
+         * if (capabilities.supportsPrimary()) {
+         * shade = shade.setPosition(capabilities, PRIMARY_ZERO_IS_CLOSED, 0);
+         * for (CoordinateSystem coordinate : coordinates) {
+         * for (int position : positions) {
+         * shade.setPosition(capabilities, coordinate, position);
+         * for (CoordinateSystem coordinate2 : coordinates) {
+         * System.out.println(String.format("%s|%s|%d|%s|%s|%s", capabilities.toString(),
+         * coordinate.toString(), position, gson.toJson(shade), coordinate2.toString(),
+         * shade.getState(capabilities, coordinate2)));
+         * }
+         * }
+         * }
+         * }
+         * }
+         */
```

Comment:
As fully commented, I propose to remove this test.

Reply:
Let me think about that. It is useful code so it may make more sense to convert ut to an actual test case.